### PR TITLE
Lint Dockerfiles with `hadolint`, not `dockerfilelint`

### LINKED ISSUE #75: Switch to using Hadolint rather than Dockerfilelint?
Further to #74, [Dockerfilelint](https://github.com/replicatedhq/dockerfilelint) is super useful, but its documented ways of disabling rules don't appear to work - unless, of course, PEBKAC! It also hasn't had a release since early 2019.

I looked around for alternatives and I found [Hadolint](https://github.com/hadolint/hadolint). It has good docs, seems to be under active development, has a bunch more stars, and as a bonus: it `brew install`s!

----

Pre-requisites:

- [x] For the Linux Actions runners to provide Homebrew. It feels weird running on macOS from the _Linuxbrew_ org. 😀 
- [ ] Potentially an audit of which Dockerfiles are still in use (can we tell)? Can we remove any outdated OS versions?
- [x] Someone to be interested/bored enough to do the work of switching us over. ✋

----

The current output that Hadolint gives on the Dockerfiles in this repo:

```shell
➜ hadolint **/Dockerfile
alpine/Dockerfile:5 DL3018 Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`
bionic/Dockerfile:6 DL3008 Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
centos6/Dockerfile:19 DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
centos7/Dockerfile:19 DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
centos8/Dockerfile:19 DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
debian10/Dockerfile:6 DL3008 Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
debian9/Dockerfile:6 DL3008 Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
fedora/Dockerfile:17 DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
opensuse/Dockerfile:16 DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
precise/Dockerfile:5 DL3008 Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
scratch-onestage/Dockerfile:7 DL3020 Use COPY instead of ADD for files and folders
scratch-onestage/Dockerfile:8 DL3020 Use COPY instead of ADD for files and folders
scratch-onestage/Dockerfile:9 DL3020 Use COPY instead of ADD for files and folders
scratch-onestage/Dockerfile:10 DL3020 Use COPY instead of ADD for files and folders
scratch-onestage/Dockerfile:11 DL3020 Use COPY instead of ADD for files and folders
scratch-onestage/Dockerfile:12 DL3020 Use COPY instead of ADD for files and folders
scratch-onestage/Dockerfile:13 DL3020 Use COPY instead of ADD for files and folders
scratch-onestage/Dockerfile:27 SC2114 Warning: deletes a system directory.
scratch/Dockerfile:1 DL3007 Using latest is prone to errors if the image will ever update. Pin the version explicitly to a release tag
travis/Dockerfile:1 DL3007 Using latest is prone to errors if the image will ever update. Pin the version explicitly to a release tag
travis/Dockerfile:5 DL3008 Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
travis/Dockerfile:65 SC2039 In POSIX sh, ulimit -n is undefined.
travis/Dockerfile:70 SC2035 Use ./*glob* or -- *glob* so names with dashes won't become options.
trusty/Dockerfile:6 DL3008 Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

```

### PR DESCRIPTION
**Pull request by issyl0**
- See commits for more details.
- Fixes #75.

(If someone else merges this, please use "rebase and merge" instead of "squash and merge" to preserve commit history. Thanks!)

### CODE REVIEW

File: centos6/Dockerfile [hunk 1]
```diff
@@ -13,8 +13,9 @@ USER linuxbrew
 WORKDIR /home/linuxbrew
 
 ENV PATH=/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH \
-	SHELL=/bin/bash \
 	USER=linuxbrew
 
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
```
Comment (sjackman) [inline centos6/Dockerfile:18]: TIL! I didn't know about this `SHELL` command.

File: travis/Dockerfile [hunk 2]
```diff
@@ -67,4 +69,4 @@ RUN umask 022
 RUN brew tap linuxbrew/xorg
 WORKDIR /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula
 # Fix error: Incorrect file permissions (664)
-RUN chmod 0644 *.rb
+RUN chmod 0644 ./*.rb
```
Comment (sjackman) [inline travis/Dockerfile]: Stylistically I prefer `chmod 0644 -- *.rb`

**Review by sjackman — APPROVED** [review]
One tiny style comment, and fantastic otherwise! This is great! Thank you, Issy.

File: travis/Dockerfile (hunk 2 above)
Comment (issyl0) [reply travis/Dockerfile]: On reflection, so do I!

**Comment by issyl0** [comment]
Thanks for the review, Shaun!

**Pull request merged**